eth/catalyst/api.go

```diff
@@ -539,6 +546,41 @@ func (api *ConsensusAPI) GetBlobsV1(hashes []common.Hash) ([]*engine.BlobAndProo
 	return res, nil
 }
 
+func (api *ConsensusAPI) GetInclusionListV1(parentHash common.Hash) (engine.InclusionList, error) {
+	if inclusionList := api.localInclusionLists.get(parentHash); inclusionList != nil {
+		return inclusionList, nil
+	}
+
+	args := &miner.BuildInclusionListArgs{
+		Parent: parentHash,
+	}
+	inclusionList, err := api.eth.Miner().BuildInclusionList(args)
+	if err != nil {
+		log.Error("Failed to build inclusion list", "err", err)
+		return nil, err
+	}
+
+	api.localInclusionLists.put(parentHash, inclusionList)
+
+	return inclusionList, nil
+}
+
+func (api *ConsensusAPI) UpdatePayloadWithInclusionListV1(payloadID engine.PayloadID, inclusionList engine.InclusionList) (*engine.PayloadID, error) {
```

Comment:
One of the functionalities of `forkchoiceUpdated` is to initiate the construction of a new payload, without updating the head of the chain.

Might be worth looking into adding the inclusion list as a parameter to `engine.PayloadAttributes`, and incorporating the functionality into `fcu` instead of expanding the engine API.

What do you think?

Reply:
That's totally doable. The main reason I've added `UpdatePayloadWithInclusionListV1` was because I wasn't sure how EL devs would feel about overloading fcu attributes.

Reply:
Yes, it's a list of unique inclusion list transactions of all observed ILs that are valid and non-equivocating ones.